### ISSUE TITLE
[BUG] Support tailwind.config.ts

### ISSUE THREAD

**Issue by itsmingjie** [francoismassart/eslint-plugin-tailwindcss#228]
**Describe the bug**
Tailwind release v3.3 today [introduced support](https://tailwindcss.com/blog/tailwindcss-v3-3) for ESM and TypeScript config files. It currently does not look like the ESLint plugin supports configuration files written in `tailwind.config.ts`. 

**To Reproduce**
1. Follow the demo instructions in the v3.3 release blog post and set up a TypeScript configuration. 
2. Run eslint
3. The configuration file is not loaded in, causing false positives. 

**Expected behavior**
Config files written with `tailwind.config.ts` should be supported. 

**Environment (please complete the following information):**
- Tailwind v3.3
- ESLint v8.36.0
- Plugin: v3.10.1

**Additional context**
This may be a duplicate of #156, but it is introduced with a different context. Escalated it to bug because there is now official support for config files written in TypeScript. 

**Related upstream PR**
https://github.com/tailwindlabs/tailwindcss/pull/10785

**Comment by taep96** [comment]
Related: https://github.com/francoismassart/eslint-plugin-tailwindcss/issues/190

I think looking up all js/ts formats is the best solution: `tailwind.config.{js,mjs,cjs,ts,mts,cts}`

You can specify the config file explicitly as a workaround for now:
```jsonc
// .eslintrc.json
{
    "extends": ["plugin:tailwindcss/recommended"],
    "settings": {
        "tailwindcss": {
            "config": "./tailwind.config.cjs"
        }
    },
    "rules": {}
```

**Comment by itsmingjie** [comment]
I think TypeScript needs to be compiled down before it can be parsed directly by the plugin

**Comment by taep96** [comment]
I was editing my comment when you said that :sweat_smile:

**Comment by pm0u** [comment]
I managed to get this working following this (somewhat ridiculous) [suggestion](https://github.com/tailwindlabs/tailwindcss/discussions/6839#discussioncomment-3678703) from before TS support existed for Tailwind.

```sh
yarn add -D @swc-node/register @swc/core
```

```js
// tailwind-config.js
const { register } = require("@swc-node/register/register")
const {
  readDefaultTsConfig,
} = require("@swc-node/register/read-default-tsconfig")
const path = require("path")
register(readDefaultTsConfig(path.join(__dirname, "tsconfig.json")))
module.exports = require("./tailwind.config").default
```

```ts
// tailwind.config.ts

export default {
  plugins: [],
  theme: {
    extend: {},
  },
  /** ... */
} 
```

```js
// .eslintrc.js

module.exports = {
  extends: [
    "plugin/tailwindcss:recommended"
  ],
  settings: {
    tailwindcss: {
      config: "./tailwind-config.js"
    }
  }
}
```

I opted to make the JS file a non-autodetectable name since some things might pick it up by default otherwise. Mainly storybook.

**Comment by tonyxiao** [comment]
Why not use `jiti` which is the strategy used by tailwind itself?

**Comment by pm0u** [comment]
I hadn't heard of jiti - i changed my implementation to use it because I was running into issues with deploying serverless functions on Vercel, they were pulling in SWC. really an issue with their dependency resolution, but since `jiti` is smaller it solved our issue so thanks for mentioning that package @tonyxiao
